homeassistant/components/zamg/config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Config Flow for zamg the Austrian "Zentralanstalt für Meteorologie und Geodynamik" integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_STATION_ID, DOMAIN, LOGGER
+from .sensor import ZamgData, closest_station
+
+
+class ZamgConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for zamg integration."""
+
+    VERSION = 1
+
+    _client: ZamgData
+
+    def _show_setup_form(
+        self,
+        user_input: dict[str, Any] | None = None,
+        errors: dict[str, Any] | None = None,
+        station_id: str = "",
+    ):
```

Comment:
This is only called from line 57 - which is enclosed in a `if user_input is None:`, so `user_input` is never a `dict` here.
I think the whole function can be removed and you can just use `self.async_show_form` in line 57 directly.
`errors` is of type `dict[str, str] | None` anyway, so no need to get an empty dict here.

Reply:
Solved out by removing the whole `_show_setup_form` function